src/SFA.DAS.ApplyService.Web/ViewModels/Roatp/TaskListViewModel.cs

```diff
@@ -48,11 +52,183 @@ public string SectionStatus(int sequenceId, int sectionId)
             return RoatpTaskListWorkflowService.SectionStatus(ApplicationSequences, NotRequiredOverrides, sequenceId, sectionId, ApplicationRouteId);
         }
 
+        public string WhosInControlCss
+        {
+            get
+            {
+                var status = WhosInControlSectionStatus;
+
+                if (status == String.Empty)
+                {
+                    return "hidden";
+                }
+
+                var cssClass = status.ToLower();
+                cssClass = cssClass.Replace(" ", "");
+
+                return cssClass;
+            }
+        }
+
+        public string WhosInControlSectionStatus
+        {
+            get
+            {
+                if (VerifiedCompaniesHouse && VerifiedCharityCommission)
+                {
+                    if ((CompaniesHouseDataConfirmed && !CharityCommissionDataConfirmed)
+                        || (!CompaniesHouseDataConfirmed && CharityCommissionDataConfirmed))
+                    {
+                        return "In Progress";
+                    }
+                    if (CompaniesHouseDataConfirmed && CharityCommissionDataConfirmed)
+                    {
+                        return "Completed";
+                    }
+                }
+
+                if (VerifiedCompaniesHouse && !VerifiedCharityCommission)
+                {
+                    if (CompaniesHouseDataConfirmed)
+                    {
+                        return "Completed";
+                    }
+                }
+
+                if (!VerifiedCompaniesHouse && VerifiedCharityCommission)
+                {
+                    if (CharityCommissionDataConfirmed)
+                    {
+                        return "Completed";
+                    }
+                }
+
+                if (!VerifiedCompaniesHouse && !VerifiedCharityCommission)
+                {
+                    if (WhosInControlConfirmed)
+                    {
+                        return "Completed";
+                    }
+                }
+
+                return "Next";
+            }          
+        }
+        
+        public string FinishCss(int sectionId)
+        {
+            var status = FinishSectionStatus(sectionId);
+
+            if (status == String.Empty)
+            {
+                return "hidden";
+            }
+
+            var cssClass = status.ToLower();
+            cssClass = cssClass.Replace(" ", "");
```

Comment:
See comments above

Reply:
CSS mapping code has been rewritten so it's clearer